dataset_loaders/parallel_loader.py

```diff
@@ -586,11 +590,17 @@ def fetch_from_dataset(self, batch_to_load):
                     seq_y = seq_y[np.newaxis, ...]
                 assert seq_y.ndim == 3
 
+            data_path_subdir = self.image_path.split(
+                self.path + '/')[1].split('/', 1)[1]
+            # Get all the filenames for the current batch to load
+            current_filenames = [fname for fname in self.filenames if
+                                 el[0][0] in fname]
```

Comment:
I am not sure if this is right. My understanding is that it's filtering the filenames with the same prefix as the one of the current batch. Is that right? If so, do you really need to do it for each batch? It seems expensive. 

If this makes sense, since we would end up passing self.path, self.shared_path and current_filenames which is also computed out of `self`, it's probably more convenient to pass `self` directly to the random_transform function (e.g., as a `dataset` argument) and access all those attributes from the function itself.